pkg/build/controller/build/build_controller.go

```diff
@@ -2334,6 +2351,53 @@ func (bc *BuildController) configMapDeleted(obj interface{}) {
 	bc.controllerConfigChanged()
 }
 
+// checkConfigMapExists checks whether all ConfigMaps specified by the Build are existing
+// if they are not, return buildUpdate for the failed build and error that led to this
+func (bc *BuildController) checkConfigMapExists(build *buildv1.Build) (*buildUpdate, error) {
+	for _, cm := range build.Spec.Source.ConfigMaps {
+		name := cm.ConfigMap.Name
+		_, err := bc.configMapClient.ConfigMaps(build.Namespace).Get(context.TODO(), name, metav1.GetOptions{})
```

Comment:
We should add a lister for the ConfigMaps to the build controller and use it here instead of a client. The build controller has a similar lister for Secrets, which comes from an informer that watches all kube objects on the cluster.